pages/book/types.mdx

```diff
@@ -13,15 +13,13 @@ Additionally, many of those types [can be made nullable](#optionals).
 
 Tact supports a number of primitive data types that are tailored for smart contract use:
 
-* [`Int{:tact}`][ints] — all numbers in Tact are $257$-bit signed integers, but [smaller representations](/book/integers#serialization) can be used to reduce storage costs.
+* [`Int{:tact}`](/book/integers) — all numbers in Tact are $257$-bit signed integers, but [smaller representations](/book/integers#serialization) can be used to reduce storage costs.
 * [`Bool{:tact}`](#booleans) — classical boolean with `true{:tact}` and `false{:tact}` values.
 * `Address{:tact}` — standard [smart contract address](https://docs.ton.org/learn/overviews/addresses#address-of-smart-contract) in TON Blockchain.
-* `Slice{:tact}`, `Cell{:tact}`, `Builder{:tact}` — low-level primitives of TON VM.
+* [`Cell{:tact}`](/book/cells#cells), [`Builder{:tact}`](/book/cells#builders), [`Slice{:tact}`](/book/cells#slices) — low-level primitives of TON VM.
 * `String{:tact}` — represents text strings in TON VM.
```

Comment:
btw, "represents text strings in TON VM" sounds confusing: TVM does not have a string primitive

Reply:
Yes, but Fift does: https://github.com/Piterden/TON-docs/blob/master/Fift.%20A%20Brief%20Introduction.md. And I didn't call it a primitive there (and in other places too), so that's that.

I can mention that Strings are stored as immutable/constant Slices though :)

Reply:
just clarify the sentence